server/protocols/jmap-rfc-8621-integration-tests/jmap-rfc-8621-integration-tests-common/src/main/scala/org/apache/james/jmap/rfc8621/contract/WebPushContract.scala

```diff
@@ -0,0 +1,508 @@
+
+    awaitAtMostTenSeconds.until(() => server.getProbe(classOf[SpoolerProbe]).processingFinished())
+  }
+
+  private def setupPushServerCallback(pushServer: ClientAndServer): AtomicReference[String] = {
+    val bodyRequestOnPushServer: AtomicReference[String] = new AtomicReference("")
+    pushServer
+      .when(request
+        .withPath(PUSH_URL_PATH)
+        .withMethod("POST"))
+      .respond(new ExpectationResponseCallback() {
+        override def handle(httpRequest: HttpRequest): HttpResponse = {
+          bodyRequestOnPushServer.set(httpRequest.getBodyAsString)
+          response()
+            .withStatusCode(HttpStatus.SC_CREATED)
+        }
+      })
+    bodyRequestOnPushServer
+  }
+
+  @Test
+  def correctBehaviourShouldSuccess(server: GuiceJamesServer, pushServer: ClientAndServer): Unit = {
+    // Setup mock-server for callback
+    val bodyRequestOnPushServer: AtomicReference[String] = setupPushServerCallback(pushServer)
+
+    // WHEN bob creates a push subscription
+    val pushSubscriptionId: String = createPushSubscription(pushServer)
+    // THEN a validation code is sent
+    awaitAtMostTenSeconds.untilAsserted { () =>
+      pushServer.verify(HttpRequest.request()
+        .withPath(PUSH_URL_PATH)
+        .withBody(json(
+          s"""{
+             |    "@type": "PushVerification",
+             |    "pushSubscriptionId": "$pushSubscriptionId",
+             |    "verificationCode": "$${json-unit.any-string}"
+             |}""".stripMargin)),
+        VerificationTimes.atLeast(1))
+    }
+
+    // GIVEN bob retrieves the validation code from the mock server
+    val verificationCode: String = Json.parse(bodyRequestOnPushServer.get()).asInstanceOf[JsObject]
+      .value("verificationCode")
+      .asInstanceOf[JsString]
+      .value
+
+    // WHEN bob updates the validation code via JMAP
+    val updateVerificationCodeResponse: String = updateValidateVerificationCode(pushSubscriptionId, verificationCode)
+
+    // THEN  it succeed
+    assertThatJson(updateVerificationCodeResponse)
+      .isEqualTo(
+        s"""{
+           |    "sessionState": "${SESSION_STATE.value}",
+           |    "methodResponses": [
+           |        [
+           |            "PushSubscription/set",
+           |            {
+           |                "updated": {
+           |                    "$pushSubscriptionId": {}
+           |                }
+           |            },
+           |            "c1"
+           |        ]
+           |    ]
+           |}""".stripMargin)
+
+    // WHEN bob receives a mail
+    sendEmailToBob(server)
+
+    // THEN bob has a stateChange on the push gateway
+    awaitAtMostTenSeconds.untilAsserted { () =>
+      pushServer.verify(HttpRequest.request()
+        .withPath(PUSH_URL_PATH)
+        .withBody(json(
+          s"""{
+             |    "@type": "StateChange",
+             |    "changed": {
+             |        "$ACCOUNT_ID": "$${json-unit.ignore-element}"
```

Comment:
```suggestion
             |        "$ACCOUNT_ID": {"Email":"$${json-unit.ignore-element}","EmailDelivery":"$${json-unit.ignore-element}","Mailbox":"$${json-unit.ignore-element}"}
```

Is likely the correct format for StateChange

CF https://jmap.io/spec-core.html#the-statechange-object

```
{
  "@type": "StateChange",
  "changed": {
    "a3123": {
      "Email": "d35ecb040aab",
      "EmailDelivery": "428d565f2440",
      "CalendarEvent": "87accfac587a"
    }, 
    // ...
  }
}
```

Reply:
For a new email, the push listener sent one message looks like:
{
  "@type": "StateChange",
  "changed": {
    "a3123": {
      "Mailbox": "d35ecb040aab"
    }, 
    // ...
  }
}

is this normal?

Reply:
Yes because the types of the push subscription is only `["Mailbox"]` (I asked myself the exact same question)